cli/agent.go

```diff
@@ -0,0 +1,90 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/spf13/cobra"
+	"go.keploy.io/server/v2/config"
+	"go.keploy.io/server/v2/pkg/agent/routes"
+	"go.keploy.io/server/v2/pkg/models"
+	"go.keploy.io/server/v2/pkg/service/agent"
+	"go.keploy.io/server/v2/utils"
+	"go.uber.org/zap"
+)
+
+func init() {
+	Register("agent", Agent)
+}
+
+func Agent(ctx context.Context, logger *zap.Logger, _ *config.Config, serviceFactory ServiceFactory, cmdConfigurator CmdConfigurator) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "agent",
+		Short: "starts keploy agent for hooking and starting proxy",
+		// Hidden: true,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			return cmdConfigurator.Validate(ctx, cmd)
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			svc, err := serviceFactory.GetService(ctx, cmd.Name())
+			if err != nil {
+				utils.LogError(logger, err, "failed to get service")
+				return nil
+			}
+
+			isdocker, _ := cmd.Flags().GetBool("is-docker")
+			enableTesting, _ := cmd.Flags().GetBool("enable-testing")
+
+			port, _ := cmd.Flags().GetUint32("port")
+			if port == 0 {
+				port = 8086
+			}
+
+			var a agent.Service
+			var ok bool
+			if a, ok = svc.(agent.Service); !ok {
+				utils.LogError(logger, nil, "service doesn't satisfy agent service interface")
+				return nil
+			}
+
```

Comment:
Don't start the server here, rather make an `agentOrchestrator` to run these two blocking commands. The server and the setup command.